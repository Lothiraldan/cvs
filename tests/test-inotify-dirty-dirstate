#!/bin/sh

# issues when status queries are issued when dirstate is dirty

"$TESTDIR/hghave" inotify || exit 80

echo "[extensions]" >> $HGRCPATH
echo "inotify=" >> $HGRCPATH
echo "fetch=" >> $HGRCPATH

echo % issue1810: inotify and fetch
mkdir test; cd test
hg init

echo foo > foo
hg add
hg ci -m foo

cd ..

hg clone test test2
cd test2
echo bar > bar
hg add
hg ci -m bar
cd ../test
echo spam > spam
hg add
hg ci -m spam
cd ../test2
hg st

# abort, outstanding changes
hg fetch -q
hg st
cd ..


echo % issue1719: inotify and mq

echo "mq=" >> $HGRCPATH

hg init test-1719
cd test-1719

echo % inserve
hg inserve -d --pid-file=hg.pid
cat hg.pid >> "$DAEMON_PIDS"

echo content > file
hg add file

hg qnew -f test.patch

hg status
hg qpop

echo % st should not output anything
hg status

hg qpush

echo % st should not output anything
hg status

hg qrefresh
hg status
