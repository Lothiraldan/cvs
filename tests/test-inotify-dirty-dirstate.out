% issue1810: inotify and fetch
adding foo
updating to branch default
1 files updated, 0 files merged, 0 files removed, 0 files unresolved
adding bar
adding spam
% issue1719: inotify and mq
% inserve
popping test.patch
patch queue now empty
% st should not output anything
applying test.patch
now at: test.patch
% st should not output anything
